enumerables.rb

```diff
@@ -0,0 +1,129 @@
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |item|
+      new_array << item if yield(item) == true
+    end
+    new_array.empty? ? self : new_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return false unless yield(item)
+      elsif pattern.nil?
+        return false unless item
+      else
+        return false unless pattern === item
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return true unless yield(item) == false
+      elsif pattern.nil?
+        return true unless item
+      else
+        return true unless (pattern === item) == false
+      end
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    new_array = []
+    my_each do |item|
+      new_array << if block_given?
+                     (yield(item) == true ? 1 : 0)
+                   elsif pattern.nil?
+                     (item == true ? 1 : 0)
+                   else
+                     ((pattern === item) == true ? 1 : 0)
+                   end
+    end
+    count = 0
+    new_array.my_each do |x|
+      count += x
+    end
+    count.zero?
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:
```ruby
def my_none?(arg = nil, &block)
!my_any?(arg, &block)
end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?